.github/workflows/ci.yml

```diff
@@ -35,10 +35,19 @@ jobs:
 
          # Use mamba for Windows dependencies
         - uses: mamba-org/setup-micromamba@v1
+          if: matrix.os == 'windows-latest' || matrix.os == 'macOS-latest'
           with:
             environment-file: ci_env.yml
             channel-priority: true
         
+        # Print the environment variables to simplify development and debugging
+        # Use conda for main dependencies
+        - uses: conda-incubator/setup-miniconda@v2
+          if: matrix.os == 'ubuntu-latest'
+          with:
+            miniforge-variant: Mambaforge
+            miniforge-version: latest
```

Comment:
I think we can use `setup-micromamba`  for both, if we need to install anything else I think we can do that via  `micromamba install`  instead of `mamba install` .